app/controllers/architectures_controller.rb

```diff
@@ -39,4 +56,55 @@ def destroy
       process_error
     end
   end
+
+  private
+
+  def can_edit?(id)
+    authorized_for(controller: :architectures, action: :update, id: id)
```

Comment:
As this is supposed to be a template, I'm wondering if it makes sense to make this generic and move that code to a concern.

```suggestion
    authorized_for(controller: controller_permission, action: model_of_controller.find_permission_name(:update), id: id)
```

Reply:
I would also like this approach, so each controller won't need to implement the permissions helper over and over.

Reply:
I moved the code to the application controller for now, and it did work!
less code for the resource controller to implement, and I am sure there are more we can refactor to make it easier.